rosdep/base.yaml

```diff
@@ -5753,6 +5753,16 @@ libsoundio-dev:
   macports: [libsoundio]
   nixos: [libsoundio]
   ubuntu: [libsoundio-dev]
+libsoup-dev:
```

Comment:
@fbe555 -  thank you for your patience, would you be willing to change this name to `libsoup2-dev`?

Per offline discussion with @clalancette & @cottsay it seems like the best compromise we could come up with for handling multiple OS's packaging different versions of 2.x and 3.x of libsoup.

Reply:
Sure - that does make better sense. 
If you wanna reward my patience, pop by my other PR #40264. I Think it's ready to merge 😉